README.md

```diff
@@ -8,7 +8,15 @@ This site is built with [Hugo](https://gohugo.io/) and served via GitHub Pages.
 
 ## Local development setup
 
-It is highly recommended to clone this repository using [`scalar`](https://git-scm.com/docs/scalar); This allows to work only on the parts of the repository relevant to your interests. You can select which directories are checked out using the [`git sparse-checkout add <directory>...`](https://git-scm.com/docs/git-sparse-checkout) command. The relevant directories are:
+It is highly recommended to clone this repository using [`scalar`](https://git-scm.com/docs/scalar); This allows to work only on the parts of the repository relevant to your interests. You can select which directories are checked out using the [`git sparse-checkout add <directory>...`](https://git-scm.com/docs/git-sparse-checkout) command. Typically, you will want to start like this:
```

Comment:
sorry, i've not been looking a lot into all the changes, and this is a bit out of scope, but: why are we recommending new contributors to use yet another (advanced) tool?

> Scalar is a repository management tool that optimizes Git for use in large repositories

when the website was just a rails app it was "simple" enough for no one even think about that. now that we have a static website do we really need **scalar**? I'd expect git to be able to cope with this simple repo without any issue

Reply:
most of the past contributors are new users who still struggle with branches and commits. adding references to scalar seems to just increase the entry barrier

Reply:
For what it's worth, I had the same thought and agree with @pedrorijo91, but decided against bringing it up in my initial review (for whatever reason, I'm not sure!).

Reply:
The problem that using `scalar` (or: partial & sparse clones, really) solves is that running Hugo and particularly Pagefind takes a really long time. In CI, Hugo spends 30+ seconds, Pagefind even 2+ minutes, to process the 25k pages (plus 45k redirects). As a contributor who wants to fix only a style issue or a typo, I find this daunting. I may be quite impatient...

Reply:
> when the website was just a rails app it was "simple" enough for no one even think about that. now that we have a static website do we really need **scalar**? I'd expect git to be able to cope with this simple repo without any issue

For the record: I never got the Rails app to work locally, not to the extent that it would render anything interesting, at least.

And yes, Git would cope with this simple, if large, repo just fine. The complete site has a lot of content, is all... last time I looked, my full `public/` folder weighed 1.2GB...

Reply:
I'm all in for quick feedback on local setup, but 3 minutes for initial setup seems very acceptable to me 😄  does it take 3 minutes for each change, or just in the first one? how long does it take to regenerate the website if we make minor changes in a couple of pages? (haven't tried it myself yet - will try to do it this weekend)

1.2GB is quite a bit indeed. my take is to have a simple setup guide. and then provide some of these tuning optimizations for more advanced contributors

Reply:
> > adding references to scalar seems to just increase the entry barrier
> 
> Would it address your concerns if I suggested "`scalar` (or `git clone --filter=blob:none --no-checkout ...`) and then `git sparse-checkout set ...`"?

I agree with @pedrorijo91, but I think having both as options is good, e.g.: "if you have `scalar` then run `[...]`, otherwise, if you have a sufficiently modern version of Git, run `[...]`, otherwise, run `[...]`".

Reply:
Heh, everyone has Scalar now, no?

Reply:
I've [updated the instructions](https://github.com/git/git-scm.com/compare/e67353ecf1a625b56bc5866c3625fe99262dc545..459cf48b6de19407b014418b66d58fee8969d741) now.

Reply:
Funny thing, I read this instruction **after** I `git-clone`d the repo, so I didn't bother to use `scalar` (confession, I've never used it). So my thought were "hey, but I've cloned it already, what now??"

Reply:
> Yes, Hugo does not really work incrementally, and Pagefind cannot work incrementally.

@dscho Isn't that what `hugo serve -w` does? When I start that it takes ~30s to build the site, and when I change a page only ~5s to update.

Reply:
> Funny thing, I read this instruction **after** I `git-clone`d the repo, so I didn't bother to use `scalar` (confession, I've never used it). So my thought were "hey, but I've cloned it already, what now??"

That's a good point @To1ne. How about something along these lines:

> [!NOTE]
> If you _already_ have a full clone and wish to accelerate development by focusing only on a small subset of the pages, you may want to run the `git sparse-checkout set [...]` command mentioned above.

> > Yes, Hugo does not really work incrementally, and Pagefind cannot work incrementally.
> 
> @dscho Isn't that what `hugo serve -w` does?

Well, yes and no. It starts out with a full build, always, and then updates incrementally _while running_ whenever the files are changed. But the initial build is never incremental, that's why I said not _really_ ;-)

> When I start that it takes ~30s to build the site, and when I change a page only ~5s to update.

Right. But that misses updates required for the Pagefind index.

Therefore, for practical purposes, I consider the build non-incremental.

Reply:
> How about something along these lines:
> 
> Note
> 
> If you _already_ have a full clone and wish to accelerate development by focusing only on a small subset of the pages, you may want to run the `git sparse-checkout set [...]` command mentioned above.

[Updated](https://github.com/git/git-scm.com/compare/459cf48b6de19407b014418b66d58fee8969d741..643b3334c805c7c56e33796f35e0aaf6821f4f37) accordingly.